client/components/Sidebar/Sidebar.jsx

```diff
@@ -0,0 +1,51 @@
+import { h } from "preact";
+import cx from "classnames";
+
+import "./sidebar.css";
+
+const BASE_CLASS = "sidebar";
+
+const Sidebar = ({ data, handleItemChange, activeIndex }) => {
+  // Get the length of the matching function
+  const getFirstFunctionMatchLength = list => {
+    if (!list.length) {
```

Comment:
It's explicit.

Would be micro-optimisation: but explicits are performant than implicit boolean typecasting.